contracts/consensus-gateway/ConsensusCogateway.sol

```diff
@@ -85,4 +94,65 @@ contract ConsensusCogateway is MasterCopyNonUpgradable, MessageBus, ConsensusGat
             _maxStorageRootItems
         );
     }
+
+    /**
+     * @notice It allows to withdraw utMOST tokens. Withdrawer needs to approve
+     *         ConsensusCoGateway contract for amount to be withdrawn.
+     *
+     * @dev Function requires :
+     *          - Amount must not be 0.
+     *          - Beneficiary must not be 0.
+     *          - Amount must be greater than gas price and gas limit.
+     *
+     * @param _amount Amount of tokens to be redeemed.
+     * @param _beneficiary The address in the origin chain where the value
+     *                     where the tokens will be released.
+     * @param _feeGasPrice Fee gas price at which reward will be calculated.
+     * @param _feeGasLimit Fee gas limit at which reward will be calculated.
+     *
+     * @return messageHash_ Message hash.
+     */
+    function withdraw(
+        uint256 _amount,
+        address _beneficiary,
+        uint256 _feeGasPrice,
+        uint256 _feeGasLimit
+    )
+        external
+        returns(bytes32 messageHash_)
+    {
+        require(
+            _amount != 0,
+            "Withdrawal amount should be greater than 0."
+        );
+        require(
+            _beneficiary != address(0),
+            "Beneficiary address must not be 0."
+        );
+        require(
+            _amount > _feeGasPrice.mul(_feeGasLimit),
+            "Withdrawal amount should be greater than max reward."
+        );
+
+        bytes32 hashWithdrawIntent = hashWithdrawIntent(
+            _amount,
+            _beneficiary
+        );
+
+        uint256 nonce = nonces[msg.sender];
+        nonces[msg.sender] = nonce.add(1);
+
+        messageHash_ = MessageOutbox.declareMessage(
+            hashWithdrawIntent,
+            nonce,
+            _feeGasPrice,
+            _feeGasLimit,
+            msg.sender
+        );
+
+        require(
+            UtilityTokenInterface(address(most)).burnFrom(msg.sender, _amount),
```

Comment:
I don't think that `UtilityTokenInterface` is needed. 
`burn` and `burnFrom` are already implemented in ERC20Token, these functions must be part of `ERC20I`

Reply:
Added burn and burnFrom in ERC20I and removed `UtilityTokenInterface`.